### DISCUSSION TITLE
Statically linking against libdbus-1

### DISCUSSION THREAD

**Discussion by schell** [diwic/dbus-rs#374] [Ideas]
I'm trying to cross compile from a number of host machines to the raspberry pi. I'm following the conversation in https://github.com/diwic/dbus-rs/blob/master/libdbus-sys/cross_compile.md. While looking for `libdbus-1.so` on my pi I saw that `libdbus-1.a` is right next to it (for me this is in `/usr/lib/arm-linux-gnueabihf/libdbus-1.a`). Are there any reasons we can't  copy this static lib and then add a `static-link` feature to `dbus-sys` to link it statically? 

Maybe my lack of knowledge about how static linking works is leading me to a false conclusion. Any help is much appreciated, and if it seems this is workable I'm willing to submit the patch.

**Comment by diwic** [comment]
Static linking is not my area of expertise either. I would argue though that if you find a way to link to `libdbus-1.a` where it resides (i e inside `/usr/lib/arm-linux-gnueabihf/`) rather than to copy the file around, it would be a better solution.

Feel free to experiment further and tell me the results!

**Comment by jirutka** [comment]
It is definitely possible. I’m building statically linked binary with dbus-rs in [connman-resolvconf](https://github.com/jirutka/connman-resolvconf/) – CI [job for x86_64](https://github.com/jirutka/connman-resolvconf/blob/7a011c2d20cc41c790bffd79c47fe6b2850a4cf6/.github/workflows/ci.yml#L129-L183), [job for aarch64](https://github.com/jirutka/connman-resolvconf/blob/7a011c2d20cc41c790bffd79c47fe6b2850a4cf6/.github/workflows/ci.yml#L217-L300).
